autoswitch_virtualenv.plugin.zsh

```diff
@@ -179,15 +179,21 @@ function check_venv()
             file_owner="$(/usr/bin/stat -f %u "$venv_path")"
             file_permissions="$(/usr/bin/stat -f %OLp "$venv_path")"
         fi
+        if [[ -d "$venv_path" ]]; then
+            default_permissions=777
```

Comment:
@MichaelAquilina : Would you be willing to accept an updated/rebased version of this PR? I'm been getting hit with this quite often recently when working with a `poetry` based env. A lot of the `poetry` commands seem to rewrite `poetry.lock`, causing its permission to be reverted to defaults. Since those `664` on my system (Ubuntu 20.04 Desktop), I get this warning every time after the file has been touched :(

Reply:
Any update on this? I've also run into the `poetry.lock` file permissions issue.

Reply:
Side note: I've been using my patch for quite a while, and it's much better, but I've still been running into the problem occasionally (not sure why and don't care enough to track down what's going on), so I'm inclined to just remove the check in my fork entirely... still think it's a useful check _in the case where `$venv_path` is an actual venv folder_ but for me, it never is.